tests/lib/eslint.js

```diff
@@ -81,6 +83,7 @@ describe("eslint", () => {
 
     afterEach(() => {
         eslint.reset();
+        eslint.rules.testReset();
```

Comment:
Instead of calling `reset` and `testReset` after each test, I think it would be better if we used a new `Linter` instance for each test. That would also verify that the instances are independent of each other. (This also applies to anywhere else we use `eslint.reset`.).